### PR TITLE
author decisions: decision logic added

### LINKED ISSUE #518: Decision action log
We need to implement logging for curator decisions in the backoffice. When a curator makes a decision on an author submission (either "Accept" or "Reject")

This task is linked with #522, please handle them together

# Tech NOTES

## Database Schema:

Create a new table named Decision to store the following fields:

- created_at: DateTime field for when the decision was made.
- updated_at: DateTime field for when the decision was made.
- user_id: id field for the curator id
- workflow_id: uuid for the workflow identifier.
- action: enum field for the action taken, corresponding to the [resolution actions](https://github.com/inspirehep/backoffice/blob/main/backoffice/backoffice/workflows/constants.py#L33)..

## Serializer:

Update the workflow serializer to include the Decision data. 

## Views
- On resolve store the decision here https://github.com/inspirehep/backoffice/blob/181b542c407bbe98e29c5d1b1bf1b4ae11e1f423/backoffice/backoffice/workflows/api/views.py#L159
- On workflows/authors  requestion we should return the decision as well:
```json
{
  "decision": "accept" | "reject" | "accept curate"
}

```

## UI
Display the decision in the UI

# NOTES:

- A Decision table is created and properly integrated into the database schema and run migrations on `dev`
- The workflow serializer is updated to include decision details.
- Curator actions are logged correctly with all required details.

### LINKED ISSUE #522: Update Resolution options for authors
We have to add `accept_curate` in the resolution options and it should run the `author_create_approved_dag` dag as well.

# TECH NOTES

* It should work with https://github.com/inspirehep/backoffice/blob/main/backoffice/backoffice/workflows/api/views.py#L159

### PR DESCRIPTION
**Pull request by DonHaul**
Besides Implementing the decision logic. I propose also two main changes in how we interact with airflow.

1. Use shared tasks, for now if two dags use the same task we were simply copy pasting.
2. Use base backoffice hook, instead of creating brand new one (else for each different model we have, a hook is created for it ) - to consider also deprecating the `workflowticket_management_hook`

### CODE REVIEW

**Comment by DonHaul** [comment]
Fixes https://github.com/cern-sis/issues-inspire/issues/522

**Comment by DonHaul** [comment]
Tests will fail as the `connection.json` were not being imported to the  github workflows

File: backoffice/backoffice/workflows/admin.py [hunk 1]
```diff
@@ -30,8 +30,41 @@ def has_permission(self, request):
         )
 
 
+class ModelAdmin(admin.ModelAdmin):
```
Comment (drjova) [inline backoffice/backoffice/workflows/admin.py]: I would rename it, not to have the same name as the admin module

File: backoffice/backoffice/workflows/api/views.py [hunk 2]
```diff
@@ -100,6 +101,38 @@ def create(self, request, *args, **kwargs):
...
+        status = self.request.query_params.get("status")
+        if status:
+            return self.queryset.filter(status__status=status)
+        return self.queryset
+
+    def create(self, request, *args, **kwargs):
+        workflow_id = request.data.get("workflow_id")
+        action = request.data.get("action")
+
+        if not all([workflow_id, action]):
```
Comment (drjova) [inline backoffice/backoffice/workflows/api/views.py]: That's why we have the serializer :) we can check if it's valid and then save

File: backoffice/backoffice/workflows/constants.py [hunk 3]
```diff
@@ -33,6 +33,10 @@ class WorkflowType(models.TextChoices):
 class ResolutionDags(models.TextChoices):
     accept = "accept", "author_create_approved_dag"
     reject = "reject", "author_create_rejected_dag"
+    accept_curate = "accept_curate", "author_create_approved_dag"
+
+
+DECISION_CHOICES = ResolutionDags.choices + []
```
Comment (drjova) [inline backoffice/backoffice/workflows/constants.py]: why `+ []`?

**Review by drjova — CHANGES_REQUESTED** [review]
Also, we since we have the `actions` for accept, reject, accept_curate, we can add the decision there. It's ok to keep the endpoint too

File: backoffice/backoffice/workflows/constants.py (hunk 3 above)
Comment (DonHaul) [reply backoffice/backoffice/workflows/constants.py]: It was so I would remember to add the decisions regarding other stuff in the future. Remove

File: backoffice/backoffice/workflows/api/views.py (hunk 2 above)
Comment (DonHaul) [reply backoffice/backoffice/workflows/api/views.py]: Reviewing. Same approach was also implemented in the `WorkflowTicketViewSet` should be reviewed as well
https://github.com/inspirehep/backoffice/blob/main/backoffice/backoffice/workflows/api/views.py#L84
Ticket created https://github.com/cern-sis/issues-inspire/issues/544

Comment (drjova) [reply backoffice/backoffice/workflows/api/views.py]: 👍 thanks for creating the issue, on django it's a best practise to use the serializer for validation

File: .github/workflows/test-workflows.yml [hunk 4]
```diff
@@ -51,6 +51,7 @@ jobs:
           -v "$(pwd)"/tests:/opt/airflow/tests
           -v "$(pwd)"/requirements-test.txt:/opt/airflow/requirements-test.txt
           -v "$(pwd)"/data:/opt/airflow/data
+          -v "$(pwd)"/scripts/connections/connections.json:/opt/airflow/connections.json
```
Comment (drjova) [inline .github/workflows/test-workflows.yml]: at the end we can mount all the scripts directory :)

File: backoffice/backoffice/workflows/api/views.py [hunk 5]
```diff
@@ -37,11 +38,21 @@
     WorkflowType,
 )
 from backoffice.workflows.documents import WorkflowDocument
-from backoffice.workflows.models import Workflow, WorkflowTicket
+from backoffice.workflows.models import Decision, Workflow, WorkflowTicket
 
 logger = logging.getLogger(__name__)
 
 
+def add_decision(workflow_id, user, action):
```
Comment (drjova) [inline backoffice/backoffice/workflows/api/views.py]: let's have a quick discussion about that :)

**Review by drjova — APPROVED** [review]

**Comment by drjova** [comment]
@DonHaul also here could you please fix the commit message? :) Is too long and we should be consistent with https://confluence.cern.ch/display/RCSSIS/Contributing+Guidelines

**Pull request merged**